docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,318 @@
+3rd party commercial tools which some of our users do. This includes but is
+not limited to security breaches through misconfigurations or 0-day 
+vulnerabilities.
+
+We are planning to close this gap by deploying the Falco runtime security
+tool.
+
+### Goals
+
+- Deploy Falco into the Gardener infrastructure.
+- Allow Gardener users to deploy Falco into their shoot clusters by adding a
+  configuration into their shoot manifests.
+- View and analyze Falco events in the Gardener dashboard.
+- Deploy an optimized ruleset for Gardener managed Kubernetes clusters based 
+  on the Falco rules [5]. Optionally allow deployment of incubating-, and
+  sandbox rules as well as custom rules.
+- Offer multiple Falco versions to users including options for manual- and 
+  automated updates.
+- Allow users to simply turn on Falco in their clusters with a reasonable 
+  default configuration as well as allow power users to fine tune the 
+  configuration (rules, plug-ins).
+- Investigate existing Falco plug-ins or develop new ones in the Gardener
+  context to broaden the detection scope
+- Provide means for users to configure Falco themselves using `falcoctl` and
+  configure their own event store.
+
+### Non-Goals
+
+We do not compete with commercial products. While we have plans to develop
+Falco plugins, our goal is not to provide a comprehensive solution provided
+by security vendors.
+
+## Proposal
+
+We are planning to provide a Gardener extension that deploys Falco and 
+Falcosidekick into shoot clusters. All events that are generated by Falco are
+sent to a central event store via Falcosidekick. Event visualization and
+analysis is provided by the Gardener dashboard.
+
+### Gardener Falco Extension
+
+Users can configure the Falco extensions by adding the following configuration
+to the `extensions` section of their shoot manifest:
+
+```
+  - type: shoot-falco-service
+    providerConfig:
+      apiVersion: falco.extensions.gardener.cloud/v1alpha1
+      kind: FalcoServiceConfig
+      # optional, will use the latest version tagged as "supported"
+      falcoVersion: 0.37.0
+      # optional, will always default to true
+      autoUpdate: true|false
+      # optional, "gardener" or "falcoctl", will default to "gardener"
+      resources: gardener
+      gardener:
+        # optional, defaults to true
+        useFalcoRules: true
+        # optional, default to false
+        useFalcoIncubatingRules: false
+        # optional, default to false
+        useFalcoSandboxRules: false
+        # references configmaps in Garden namespace containing custom rules
+        ruleRefs:
+        - ref: rules1
+      falcoctl:
+        # Falcoctl configuration as defined in 
+        # https://github.com/falcosecurity/charts/blob/master/charts/falco/values.yaml#L366
+      # publish Falco events to other event store (optional)
+      customWebhook:
+        enabled: true|false
+        address: <http...>
+        customheaders: ...
+        checkcerts: ...
```

Comment:
what do you mean with `checkcerts`?

Reply:
This is from the [Falcosidekick helm configuration](https://github.com/falcosecurity/charts/blob/master/charts/falcosidekick/values.yaml#L473). I have added a note.